### PR TITLE
fix: Typescript fixes

### PR DESCRIPTION
**Pull request by SimpleCreations**
# Overview

* Fixes issues described in #118 
* Defines colors as `ColorValue` instead of `string`

### CODE REVIEW

**Comment by mmazzarolo** [comment]
Thank you @SimpleCreations ! 🧡
I'm taking a look at it, mind fixing the linting error in the meanwhile? (Otherwise, I'll fix it later — not sure why Husky didn't fix it)

File: src/Container.tsx [hunk 1]
```diff
@@ -1,87 +1,91 @@
...
-        buttonChildrens.push(
-          <View
-            style={[
-              verticalButtons
-                ? styles.buttonSeparatorVertical
-                : styles.buttonSeparatorHorizontal,
-              buttonSeparatorStyle,
-            ]}
-          />
-        );
```
Comment (mmazzarolo) [inline src/Container.tsx:79]: Wow, I think I haven't [read this code in ages](https://github.com/mmazzarolo/react-native-dialog/blob/8cc69b78cf391a39994abc6774b439ac29622a9f/src/Container.js), it's 4 years old — no idea why the strings were hardcoded in the first place, haha. 
Thank you so much for cleaning this up!

**Review by mmazzarolo — COMMENTED** [review]
Code is looking solid.  
Thank you for going above and beyond the TypeScript fixes!  

I'll give it a quick run in a couple of hours and then I'll release it. 

Wondering if this should be a major bump or not. 
At a first glance, I'm not seeing cases where we're making existing types more strict. But normally, when we do these many type changes in a single shot, I prefer doing major releases to ensure we don't break existing build type-checks. 

What do you think?

**Comment by SimpleCreations** [comment]
> Wondering if this should be a major bump or not.
> At a first glance, I'm not seeing cases where we're making existing types more strict. But normally, when we do these many type changes in a single shot, I prefer doing major releases to ensure we don't break existing build type-checks.
> 
> What do you think?

I don't see a major version bump being necessary because yes, seems like we're only making the types less strict.

However I don't have experience maintaining public npm packages so I'm not sure about the potential pitfalls.

**Review by mmazzarolo — APPROVED** [review]

**Pull request merged**